Solutions API changes

junaid-ali opened the pull request.

Commit:
Solutions API changes

Signed-off-by: plumgrid <Junaid Ali>

Commit:
Changes:
  - Removed is_zone_info_available method
  - Code optimization

Signed-off-by: plumgrid <Junaid Ali>

junaid-ali commented:
@JaveriaK @bilalbaqar @salmankh @MQasimSarfraz @akhayam

syedalijabir commented on hooks/pg_dir_utils.py:
```diff
@@ -398,6 +411,141 @@ def post_pg_license():
+    print "POST_IPS    {}".format(status)
+    if 'success' in _exec_cmd_output(
+        status,
+            'No response from specified LCM IP!'):
```
No response from SAPI to be exact, LCM has nothing to do with it.

junaid-ali replied:
The issue can be due to both LCM and Solutions API port.

syedalijabir commented on hooks/pg_dir_utils.py:
```diff
@@ -398,6 +411,141 @@ def post_pg_license():
+    print "POST_LICENSE status: {}".format(status)
+    if 'success' in _exec_cmd_output(
+        status,
+            'No response from specified LCM IP!'):
```
Same, user should see the reply from Solutions API

syedalijabir commented on hooks/pg_dir_utils.py:
```diff
@@ -398,6 +411,141 @@ def post_pg_license():
+
+def sapi_post_zone_info():
+    sol_name = '"solution_name":"Ubuntu OpenStack"'
+    sol_version = '"solution_version":"12"'
```
Needs to be a variable for kilo/liberty/mitaka etc.

syedalijabir commented on hooks/pg_dir_utils.py:
```diff
@@ -398,6 +411,141 @@ def post_pg_license():
+    print "ZONE_INFO status = {}".format(status)
+    if 'success' in _exec_cmd_output(
+        status,
+            'No response from specified LCM IP!'):
```
Same.

syedalijabir commented on hooks/pg_dir_utils.py:
```diff
@@ -398,6 +411,141 @@ def post_pg_license():
+        if "success" in status:
+            return True
+    except:
+        log('No response from specified LCM IP')
```
Same.

bilalbaqar commented on actions.yaml:
```diff
@@ -0,0 +1,8 @@
+restart-pg:
+  description: Restart the plumgrid-director unit's service.
+post-ips:
```
As actions are available to the end user dont you think these names should be more descriptive? What if we add a prefix like sapi-post-ips so that the user does not confuse this action for something else.

junaid-ali replied:
Noted

bilalbaqar commented on hooks/pg_dir_context.py:
```diff
@@ -26,6 +26,30 @@
+    ips of the edge nodes
+    '''
+    return [get_host_ip(rdata['private-address'])
+            for rid in relation_ids("plumgrid")
```
this would also get the gateway IPs right?

bilalbaqar commented on hooks/pg_dir_hooks.py:
```diff
@@ -117,13 +124,17 @@ def config_changed():
     if charm_config.changed('plumgrid-virtual-ip'):
         CONFIGS.write_all()
+        for rid in relation_ids('plumgrid'):
+            plumgrid_joined(rid)
```
I believe plumgrid_joined only sets the opsvm IP in the relation. Why would you need it here?

Commit:
Changes:

 - Renamed actions related to Solutions API
 - Added openstack-release config to provide solution version

Signed-off-by: plumgrid <Junaid Ali>
